### ISSUE TITLE
webpack-dev-server not found

### ISSUE THREAD

**Issue by mprogers** [PacktPublishing/React-and-React-Native#2]
Following the instructions in Instruction/README, I ran 
npm install (in the root level, REACT-AND-REACT-NATIVE-MASTER).
npm install -g webpack
cd'd to Chapter02/builtin-html-tags
webpack-dev-server

and got, for my pains:
-bash: webpack-dev-server: command not found

What am I doing wrong?

**Comment by evadeflow** [comment]
This appears to be a small error in the book's text: `webpack-dev-server` is a separate package. It should have been installed—*locally*—when you ran `npm install` from the top-level directory. So you might try running it this way instead:

```
$ node ../../node_modules/webpack-dev-server/bin/webpack-dev-server.js
```

This 'works' for me in the sense that it at least *runs*. But now I'm having the same problem described in Issue #1, namely: "there is no browser output for `index.html`". I hit `localhost:8080` in my browser, and I see the title `Builtin HTML Tags` in the tab for the page, but... the page itself is entirely blank. (I guess I'll add my $0.02 to #1 now...)

**Comment by JasonPrime8** [comment]
Experiencing similar problems.  Try the following, this seems to be working for me:

1) Install webpack **both globally and locally**, i.e. at the base directory of the code do the following:

```
npm install webpack --global
npm install webpack --save-dev
```

2) Install the webpack dev server, **also both globally and locally**:

```
npm install webpack-dev-server --global
npm install webpack-dev-server --save-dev
```

3) Update the webpack.config.js files, **in particular the "loader" line**, as follows:

    (A) Replace 'react-hot' with 'react-hot-loader/webpack'

    (B) Replace 'babel' with 'babel-loader'

So for example, the "hello-jsx" file here is the line in the OLD, ORIGINAL webpack.config.js file:

```
loaders: ['react-hot', 'babel?presets[]=es2015'],
```

And here is the line in the NEW, UPDATE webpack.config.js file:

```
loaders: ['react-hot-loader/webpack', 'babel-loader?presets[]=es2015'],
```
4) Run the webpack dev server (from the subdirectory) **with the hot and inline options**, e.g.

```
$ pwd
[whatever_base_directory]/Chapter02/hello-jsx
$ webpack-dev-server --hot --inline
```

Again, this is what I found doing some digging and reading...things seem to be working for me (first few examples that I've tried it on), was able to then go to browser and run app at localhost:8080...let me know if works for others!

**Comment by evadeflow** [comment]
@Prime8Labs, thanks for your help—it works! I didn't wind up needing to install `webpack` and `webpack-dev-server` globally, though.  I simply needed to add `--hot` to my `webpack-dev-server` invocation (i.e., `node ../../node_modules/webpack-dev-server/bin/webpack-dev-server.js --hot`) and change the 'loaders' line in `webpack.config.js` to `loaders: ['react-hot-loader/webpack', 'babel?presets[]=es2015'],`.

**Comment by adamboduch** [comment]
@mprogers Thanks for the report. I'm sorry that you're off to a rough start. Let's fix that.

@evadeflow Thanks for the tips.

@JasonPrime8 Thanks for all the digging.

So to fix this, I've done a couple things. First, I've updated the [README](https://github.com/PacktPublishing/React-and-React-Native/blob/master/README.md) to include things like the global packages necessary to run the examples. In general, it's just easier to get from nothing to running the first example. So if you want to blow away your environment and pull this repo, things should be much cleaner.

Second, I've started to simplify all of the Webpack configs in the examples. We're using Webpack 2 now, so there's far fewer cryptic messages when something does go wrong. For example, you shouldn't _need_ to provide `--inline` or `--hot` anymore.

Thanks again for your feedback and patience.

58bce912285801d2975f57242afe158866662aec
